aptos-move/aptos-vm-types/src/resource_group_adapter.rs

```diff
@@ -78,12 +78,22 @@ impl<'r> ResourceGroupAdapter<'r> {
     ) -> Self {
         let group_size_kind = GroupSizeKind::from_gas_feature_version(
             gas_feature_version,
-            resource_group_charge_as_size_sum_enabled,
+            // Even if flag is enabled, if we are in non-capable context, we cannot use AsSum,
+            // and split resource groups in the VMChangeSet.
+            // We are not capable if:
+            // - Block contains single PayloadWriteSet::Direct transaction
+            // - we are not executing blocks for a live network in a gas charging context
+            //     (outside of BlockExecutor) i.e. unit tests, view functions, etc.
+            //     In this case, disabled will lead to a different gas behavior,
+            //     but gas is not relevant for those contexts.
+            resource_group_charge_as_size_sum_enabled
+                && maybe_resource_group_view
+                    .map_or(false, |v| v.is_resource_group_split_in_change_set_capable()),
         );
 
         Self {
             maybe_resource_group_view: maybe_resource_group_view.filter(|view| {
-                view.is_resource_group_split_in_change_set_enabled()
+                view.is_resource_group_split_in_change_set_capable()
```

Comment:
given the logic of setting AsSum, this can be asserted it seems if AsSum is true and there is a view?